riptide-faults/src/main/java/org/zalando/riptide/faults/DefaultFaultClassifier.java

```diff
@@ -14,13 +14,12 @@ public DefaultFaultClassifier(final Predicate<Throwable> isTransient) {
 
     @Override
     public Throwable classify(final Throwable throwable) {
-        final Throwable root = Throwables.getRootCause(throwable);
-
-        if (isTransient.test(root)) {
-            return new TransientFaultException(throwable);
-        }
-
-        return throwable;
+        return Stream.iterate(throwable, Throwable::getCause)
+                .filter(Objects::nonNull)
+                .findFirst()
```

Comment:
Looks like `.findFirst()` and `.filter(isTransient)` should be swapped

Reply:
That actually will break if a non-transient throwable is given without any cause.

Reply:
Why? iterate return throwable itself as first element

Reply:
Yes, but if it's not transient then the stream is empty and findFirst will throw.

Reply:
> Returns an Optional describing the first element of this stream, or an empty Optional if the stream is empty.

Reply:
I think I misinterpreted an exception that I saw during tests.